plugins/Web/plugin.py

```diff
@@ -149,19 +149,31 @@ def noIgnore(self, irc, msg):
     def getTitle(self, irc, url, raiseErrors):
         size = conf.supybot.protocols.http.peekSize()
         timeout = self.registryValue('timeout')
-        (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
+        try:
+            (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
                 timeout=timeout)
+        except Exception as e:
+            if raiseErrors:
+                irc.error(_('That URL raised <' + str(e)) + '>',
+                          Raise=True)
+                return None
+            elif not raiseErrors:
+                self.log.info('Web plugin TitleSnarfer: URL <%s> raised <%s>', url, str(e))
+                return None
         try:
             text = text.decode(utils.web.getEncoding(text) or 'utf8',
                     'replace')
         except UnicodeDecodeError:
-            pass
-        if minisix.PY3 and isinstance(text, bytes):
-            if raiseErrors:
-                irc.error(_('Could not guess the page\'s encoding. (Try '
-                        'installing python-charade.)'), Raise=True)
-            else:
-                return None
+            if minisix.PY3:
+                if raiseErrors:
+                    irc.error(_('Could not guess the page\'s encoding. (Try '
+                                'installing python-charade.)'), Raise=True)
+                    return None
+                elif not raiseErrors:
```

Comment:
I would condense this into `else:`, since a boolean can only be either True or False.

Reply:
Hi jlu5, thanks. Changed for all similar occurrences.